frontend/appflowy_flutter/lib/plugins/database/widgets/cell/editable_cell_skeleton/url.dart

```diff
@@ -135,6 +144,83 @@ class _GridURLCellState extends GridEditableTextCell<EditableURLCell> {
   String? onCopy() => cellBloc.state.content;
 }
 
+class MobileURLEditor extends StatelessWidget {
+  const MobileURLEditor({
+    super.key,
+    required this.bloc,
+    required this.textEditingController,
+  });
+
+  final URLCellBloc bloc;
+  final TextEditingController textEditingController;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        const VSpace(
+          4.0,
+        ),
+        Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 16),
+          child: FlowyTextField(
+            controller: textEditingController,
+            hintStyle: Theme.of(context)
+                .textTheme
+                .bodyMedium
+                ?.copyWith(color: Theme.of(context).hintColor),
+            hintText: LocaleKeys.grid_url_textFieldHint.tr(),
+            textStyle: Theme.of(context).textTheme.bodyMedium,
+            keyboardType: TextInputType.url,
+            hintTextConstraints: const BoxConstraints(maxHeight: 52),
+            onChanged: (_) {
+              if (textEditingController.value.composing.isCollapsed) {
+                bloc.add(URLCellEvent.updateURL(textEditingController.text));
+              }
+            },
+            onSubmitted: (text) => bloc.add(URLCellEvent.updateURL(text)),
+          ),
+        ),
+        const VSpace(
+          8.0,
+        ),
+        MobileQuickActionButton(
+          onTap: () {
+            openUrlCellLink(textEditingController.text);
+            context.pop();
+          },
+          icon: FlowySvgs.url_s,
+          text: LocaleKeys.grid_url_launch.tr(),
+        ),
+        const Divider(
+          height: 8.5,
+          thickness: 0.5,
+        ),
+        MobileQuickActionButton(
+          onTap: () {
+            if (textEditingController.text.isNotEmpty) {
+              Clipboard.setData(
+                ClipboardData(text: textEditingController.text),
+              );
+            }
+            Fluttertoast.showToast(
+              msg: LocaleKeys.grid_url_copiedNotification.tr(),
+              gravity: ToastGravity.BOTTOM,
+            );
```

Comment:
If the URL field is empty, we won't setData, but we will show a toast, so the Clipholder will still be filled with whatever was there before.

We can consider: 1. Disabling the button if the text is empty, 2. Show a message saying there was nothing to copy, 3. set the clipholder data to empty anyway and show the toast

Reply:
Thanks I totally glossed over this. I disable the button as well for the visit link button